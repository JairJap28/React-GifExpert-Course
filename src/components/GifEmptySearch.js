import React from 'react';

const EmptySearch = () => {
    return (
        <div>
            <p>There is no data to show yet, please enter a valid keyword.</p>
        </div>
    );
}

export default EmptySearch;