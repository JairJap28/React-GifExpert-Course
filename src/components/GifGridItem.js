import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <img className="img__normal" src={url} alt={title}/>
            <img className="img__blur" src={url} alt={`blur ${title}`}/>
            <p>{title || 'No title description'}</p>
        </div>
    )
};

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}