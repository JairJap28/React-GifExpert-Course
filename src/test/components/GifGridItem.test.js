import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('Test in <GifGridItem>', () => {
    const title = 'A Title';
    const url = 'https://localhost/image.png';
    const wrapper = shallow(< GifGridItem title={title} url={url}/>);

    test('Should show the text correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should have a paragraph with the title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('Should have the image src and alt equals to the props', () => {
        const img = wrapper.find('.img__normal');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    }); 

    test('Should have the class animate__fadeIn', () => {
        const div = wrapper.find('.card');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    });
});
