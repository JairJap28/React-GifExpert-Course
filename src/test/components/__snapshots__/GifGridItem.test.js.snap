// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem> Should show the text correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A Title"
    className="img__normal"
    src="https://localhost/image.png"
  />
  <img
    alt="blur A Title"
    className="img__blur"
    src="https://localhost/image.png"
  />
  <p>
    A Title
  </p>
</div>
`;
